### PR TITLE
319-bulk-student-validation

### PR DESCRIPTION
**Pull request by danhalson**
## Status

- Closes https://github.com/RaspberryPiFoundation/digital-editor-issues/issues/319

## What's changed?

- Handles validations from profile when creating bulk students

## Steps to perform after deploying to production

N/A

### CODE REVIEW

File: spec/features/school_student/creating_a_batch_of_school_students_spec.rb [hunk 1]
```diff
@@ -46,17 +63,17 @@
     expect(ProfileApiClient).not_to have_received(:create_safeguarding_flag).with(token: UserProfileMock::TOKEN, flag: ProfileApiClient::SAFEGUARDING_FLAGS[:teacher], email: owner.email)
   end
 
-  it 'responds 204 No Content' do
+  it 'responds 202 No Content' do
```
Comment (conorriches) [inline spec/features/school_student/creating_a_batch_of_school_students_spec.rb:66]:
```suggestion
  it 'responds 202 Accepted' do
```

File: spec/factories/user_job.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :user_job do
+    association :good_job, factory: :good_job
+    user_id { SecureRandom.uuid } # Assuming user_id is a UUID
```
Comment (conorriches) [inline spec/factories/user_job.rb:6]: Just want to check this assumption that user_id is a UUID still stands? Can this comment be removed?

**Review by conorriches — COMMENTED** [review]
Looking very nice, I only have a couple of thoughts but this looks great :D

File: spec/features/school_student/creating_a_batch_of_school_students_spec.rb (hunk 1 above)
Comment (danhalson) [reply spec/features/school_student/creating_a_batch_of_school_students_spec.rb]: Good spot on this and the other, removed! thanks

**Pull request merged**